src/Toolkit/Toolkit.WPF/UI/Controls/PopupViewer/PopupMediaView.cs

```diff
@@ -0,0 +1,203 @@
+﻿// /*******************************************************************************
+//  * Copyright 2012-2018 Esri
+//  *
+//  *  Licensed under the Apache License, Version 2.0 (the "License");
+//  *  you may not use this file except in compliance with the License.
+//  *  You may obtain a copy of the License at
+//  *
+//  *  http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  *   Unless required by applicable law or agreed to in writing, software
+//  *   distributed under the License is distributed on an "AS IS" BASIS,
+//  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  *   See the License for the specific language governing permissions and
+//  *   limitations under the License.
+//  ******************************************************************************/
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping.Popups;
+using Esri.ArcGISRuntime.Toolkit.Internal;
+using Esri.ArcGISRuntime.UI;
+using System.IO;
+using System.Windows.Input;
+using System.Windows.Media.Imaging;
+using System.Xaml;
+
+namespace Esri.ArcGISRuntime.Toolkit.Primitives
+{
+    /// <summary>
+    /// Supporting control for the <see cref="Esri.ArcGISRuntime.Toolkit.UI.Controls.PopupViewer"/> control,
+    /// used for rendering a <see cref="PopupMedia"/>.
+    /// </summary>
+    public class PopupMediaView : ContentControl
+    {
+        private double _lastChartSize = 0;
+        private const double MaxChartSize = 1024;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PopupMediaView"/> class.
+        /// </summary>
+        public PopupMediaView()
+        {
+            HorizontalContentAlignment = HorizontalAlignment.Stretch;
+            VerticalContentAlignment = VerticalAlignment.Stretch;
+        }
+
+        /// <inheritdoc />
+        protected override void OnDpiChanged(DpiScale oldDpi, DpiScale newDpi)
+        {
+            _lastChartSize = 0;
+            base.OnDpiChanged(oldDpi, newDpi);
+        }
+
+        /// <inheritdoc />
+        protected override Size MeasureOverride(Size constraint)
+        {
+            if (PopupMedia != null && PopupMedia.Type != PopupMediaType.Image)
+            {
+                UpdateChart(constraint);
+            }
+            return base.MeasureOverride(constraint);
+        }
+
+        private void UpdateMedia(Size desiredSize)
+        {
+            if (PopupMedia is null || PopupMedia.Value is null)
+            {
+                Content = null;
+                return;
+            }
+            if (PopupMedia.Type == PopupMediaType.Image)
+            {
+                UpdateImage();
+            }
+            else
+            {
+                UpdateChart(desiredSize);
+            }
+        }
+
+        private void UpdateImage()
+        {
+            if(PopupMedia is null)
+            {
+                Content = null;
+            }
+            else if (PopupMedia.Type == PopupMediaType.Image)
```

Comment:
The logic for getting a usable image out of a PopupMediaValue seems pretty complex.  Is toolkit the right place for this? Perhaps we should provide something like "ToImageSource()" extension right in dotnet-api, the way we did for RuntimeImage?

Reply:
Yes but out of scope of this PR